src/wakefs/fuse.cpp

```diff
@@ -217,16 +233,56 @@ bool run_in_fuse(fuse_args &args, int &status, std::string &result_json) {
   (void)close(STDOUT_FILENO);
   (void)close(STDERR_FILENO);
 
-  do waitpid(pid, &status, 0);
-  while (WIFSTOPPED(status));
+  pid_t timeout_pid = -1;
+
+  // Launch timer process
+  if (args.command_timeout) {
+    timeout_pid = fork();
+    if (timeout_pid < 0) {
+      std::cerr << "wakebox: failed to fork timeout process" << std::endl;
+      exit(1);
+    }
+
+    if (timeout_pid == 0) {
+      prctl(PR_SET_NAME, "wb-timer", 0, 0, 0);
+      sleep(*args.command_timeout);
+      exit(124);
+    }
+  }
+
+  pid_t wait_pid;
+  while ((wait_pid = wait(&status)) != -1) {
+    if (wait_pid == timeout_pid && WIFEXITED(status)) {
+      kill(payload_pid, SIGKILL);
+
+      struct timeval stop;
+      gettimeofday(&stop, 0);
+      std::string output;
+      args.daemon.disconnect(output);
+      RUsage usage = {};
```

Comment:
It's probably nice to report the real usage back regardless of the timeout, rather than `{}`, it also covers non-time usage like file io bytes written

Reply:
There isn't actually a good way to get the real info out. RUsage reports back the usage for all terminated child processes that have been `wait()`ed on. We could wait on the payload process after killing it but at that point the timeout process would have terminated+waited so the usage would be higher than expected. 

My thought was that it was better to give no info than to give incorrect info. But I don't feel strongly either way

Reply:
I suppose seeing the usage on a timeout could be helpful for debugging. Like "oh yeah, that was writing massive amounts out to the disk and it was using 10x more memory than we wanted it to"

Reply:
Unless there is significant push back/disagreement, I'd like to just land this PR since its been dragging on this we can make a separate one to explore usage

Reply:
I'm happy to land. I don't think the usage is required.

Reply:
We should at least put in `runtime` which we can just copy straight in from the timeout value

Reply:
We still insert runtime. That is derived from `gettimeofday(&stop, 0);` not `usage`

Reply:
```
$ cat result.json | jq                
{
  "usage": {
    "status": 124,
    "membytes": 0,
    "inbytes": 0,
    "outbytes": 0,
    "runtime": 1.001355,
    "cputime": 0
  },
  "inputs": [],
  "outputs": [],
  "timed-out": true,
  "run-host": "ashleycoleman-1518"
}
```

Reply:
ah ok